### ISSUE TITLE
Virtualmin Package Updates - Improvements

### ISSUE THREAD

**Issue by shoulders** [virtualmin/virtualmin-gpl#861]
|  SYSTEM INFORMATION||
|----------------------|---------------------------|
|  OS type and version |     Ubuntu Linux 22.04.4     |
|  Webmin version      |     2.111      |
|  Usermin version     |     2.010     |
|  Virtualmin version  |     7.10.0       |
|  Theme version       |     21.10      |
|  Package updates     |      108  package updates are available   |

### The background

This is my `Virtualmin Package Updates`

![image](https://github.com/user-attachments/assets/dd47c138-9eab-4938-a43c-709ef69fb9a8)

### the issues

- This list can get very long (NB: as an admin I should not let it get this long, but it happens 😄). 
- is the `Install All Updates Now` a bit dangerous as it just encourages people to blindly click the button without checking what they are updating. I am not sure how many people will inspect this list before pressing the button.
- the statement `83 updates to Virtualmin packages are available.` is wrong. Joe always advices the community that Webmin/Virtualmin sits on top of your OS, it is not the OS, I think this is one of his pet hates.
- The title `Virtualmin Package Updates` is not a correct name for this module for the reasons mentioned above.

### Proposed Solutions

- rename module to `OS Package Updates`
- remove the parsed list of updates and replace with a summary (see image below)
- `83 updates to Virtualmin packages are available.` this will not be required as when using a summary format (see image below)
- use a traffic light system for available updates (see image below)
- change `Install All Updates Now` to `Goto Software Package Updates`

![image](https://github.com/user-attachments/assets/2b5ce3ce-4365-4803-bb62-5686c561deda)

![image](https://github.com/user-attachments/assets/0ddb3897-6a78-44d9-bdf3-2cd6cf8838b0)

Green = no updates available
Orange = Regular Updates only available
Red = Updates contain security updates

**Comment by iliajie** [comment]
Hello,

Thanks for the heads up!

> A bit dangerous as it just encourages people to blindly click the button without checking what they are updating.

I thought those listed packages were pretty clear about what is going to be installed.



> * the statement `83 updates to Virtualmin packages are available.` is wrong. Joe always advices the community that Webmin/Virtualmin sits on top of your OS, it is not the OS, I think this is one of his pet hates.

This is a good point! Jamie, perhaps we deviated from the original intention to list and install only Virtualmin-related packages in this part of the UI? If not, we should fix the message and the accordion name itself. But I think it would be better to install only Virtualmin-related packages rather than projecting a list of all available packages.



> * The title `Virtualmin Package Updates` is not a correct name for this module for the reasons mentioned above.

Agreed!



> * Rename module to `OS Package Updates`

This functionality better fits the "Software Package Updates" module itself. I think the Virtualmin module should only return the list of Virtualmin-related packages.



> * use a traffic light system for available updates (see image below)

There is already one "traffic light" on the dashboard that alerts about available package updates. I don't think adding counts helps anyone. Though, it might be cool to include this feature in the "Software Package Updates" module itself.

**Comment by shoulders** [comment]
> I thought those listed packages were pretty clear about what is going to be installed.

Yes the sit shows what is going to be installed, but how many new users will read them and then find that one update they installed broke something in theier server. I grant you this might be a rare thing and better admins would check the list before hitting the button, My logic is by forcing the user to the software module it will encourage them to check the updates before applying them (i.e. virtulmin updates that might have issues and you are told not to install them). This was some of my thinking behind it.

**Comment by shoulders** [comment]
> There is already one "traffic light" on the dashboard that alerts about available package updates. 

![image](https://github.com/user-attachments/assets/006f6584-4358-408c-a67c-cc0d11d4dd3a)

Does this actually change colour depending on what type of updates are available, I am sure it is only ever red.

**Comment by shoulders** [comment]
further to above, do you mean that the virtualmin is highlighted as shown below when an update is available.

![image](https://github.com/user-attachments/assets/25e1a51b-a77c-4039-bd41-f86705af19ef)

**Comment by iliajie** [comment]
> Yes the list shows what is going to be installed, but how many new users will read them and then find that one update they installed broke something in their server.

Clicking the same button later to install all or selected list of packages would do it either way. I don't think it adds any protection against potentially corrupt packages.

> I grant you this might be a rare thing and better admins would check the list before hitting the button.

Yeah, almost never happens.

> Does this actually change colour depending on what type of updates are available, I am sure it is only ever red.

Yes, red only is fine. No need to make it a rainbow, even though I love rainbows.


> further to above, do you mean that the virtualmin is highlighted as shown below when an update is available.

I don't understand the question.

**Comment by iliajie** [comment]
> When you refer to the traffic light system do you mean

I really don't know! 🚦 I just reused the term you initially used and put it in quotes:

> > * use a traffic light system for available updates (see image below)
> 
> There is already one "traffic light" on the dashboard that alerts about available package updates.

Anyway, thanks for the heads up! This is a good suggestion to only list Virtualmin-related packages. I will look into fixing this part of the UI.

**Comment by shoulders** [comment]
Might just be a British thing

Traffic light system to me implies the colours green, yellow and red are used, like they do on food for an example. But I think in this case you just mean highlighting.

![image](https://github.com/user-attachments/assets/224af2e9-e107-4f38-bf87-0f05b8c7032e)

Thanks for looking in to this, as always 😄

**Comment by jcameron** [comment]
FYI, "virtualmin package updates" covers both packages supplied by Virtualmin itself as well as packages related to the servers we manage (Apache, MariaBD, etc). Other OS packages are excluded...

**Comment by iliajie** [comment]
Shouldn't we make it clearer and only display packages related to Webmin and Virtualmin, i.e., packages from our repos?

**Comment by jcameron** [comment]
We certainly could! I guess it's a question of what makes most sense to the user ... my feeling was that we want to encourage and prioritize all packages related to Virtualmin above general OS updates.

@swelljoe what do you think?

**Comment by iliaross** [comment]
Reopened, as this was closed by an automated cleanup run. @shoulders, sorry for the noise, and thanks for the contribution. We definitely didn't mean any disrespect.

If it turns out the issue is already resolved for you, you're welcome to close it.
